### ISSUE TITLE
The matching sequence does not exist in the custom DB

### ISSUE THREAD

**Issue by juejun14** [soedinglab/MMseqs2#676]
Hello,
thank you for creating this tool.

I have a question about the taxonomy assignment.
I have used the command "mmseqs filtertaxseqdb" to filter the on interest sequence by taxon ID.
It  works, but I realize that there are some sequences matched which are not exist in my DB.
For exemple, I use uniref90 DB and I would like to remove all the sequences of "Candidatus", I have written a script to extract taxon ID from the uniref90.fsa, and then I run  
```bash 
mmseqs filtertaxseqdb UniRef90 mmseq2DB/uniref90_cleaned_DB --taxon-list "!743074&&!1986619&&!2013779&&!1256583&&!1618594&&!1974011&&!2546570&&!2719382"
```
to filter my DB.

I have found that there are some matched with species"Candidatus Helarchaeota archaeon" (taxID=2719382) when I checked the report file.
I have check the sequence of this species is not exist in Uniref90.fsa. but it exists in the database file "uniref90_cleaned_DB_h".

Does anyone know why mmseqs taxonomy can match the sequence that don't exist in DB? 
And is there any way to remove also these sequences when using taxonomy assignment?

Thanks,

**Comment by milot-mirdita** [comment]
I think you need an OR (`||`) between the taxids. I didn't try it out, but using OR should work correctly. 
The statement will be: Keep the sequence if its NOT 743074 OR NOT 1986619...

**Comment by juejun14** [comment]
> I think you need an OR (`||`) between the taxids. I didn't try it out, but using OR should work correctly. The statement will be: Keep the sequence if its NOT 743074 OR NOT 1986619...

Thank you for your reply, I have already tried " || " at the first time, but it doesn't work to remove the sequence I want to remove. So I have tried && and it works well. 
I have compared the taxonomy report between the DB and filtered DB, it has showed also that I have already removed the no interest taxon in DB. (by use &&)

The problem is that the report have showed some taxon which are not exist in my DB even though I don't filter it...
So I don't think the problem is with the filter

**Comment by milot-mirdita** [comment]
Can you write down the steps you execute after the taxonomy filtering? I am not sure how to reproduce the issue.

**Comment by milot-mirdita** [comment]
I don't think that "Candidatus Izimaplasma" is part of any of the 8 taxonomic clades that you filtered out.

I don't have a UniRef90 set-up to try to reproduce your issue directly. I think what you would likely need is a way to filter for string(s) instead of filtering by taxonomic identifiers. Would that help?

Also I am not sure it makes sense to remove all Candidatus in general. I don't know your specific application but Candidatus just means there is no cultured type strain, but the species is usually still well described.

**Comment by juejun14** [comment]
> I don't think that "Candidatus Izimaplasma" is part of any of the 8 taxonomic clades that you filtered out. I don't have a UniRef90 set-up to try to reproduce your issue directly. I didn't check for "Candidatus Helarchaeota archaeon" (taxID=2719382) because of this.
> 
> I think what you would likely need is a way to filter for string(s) instead of filtering by taxonomic identifiers. Would that help?
> 
> Also I am not sure it makes sense to remove all Candidatus in general. I don't know your specific application but Candidatus just means there is no cultured type strain, but the species is usually still well described.

I did filter by string before using taxon id, and i created the DB from the filtered fasta file. But I found that the DB I created has removed some important nodes  so that I cannot use for taxonomy assignment.

Finally I found out why my filter didn't work, it related to the version of Uniref90, I have retried with a latest uniref90 set-up, it works!

Thanks anyway and have a nice day
